sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryRelation.scala

```diff
@@ -237,7 +238,23 @@ case class CachedRDDBuilder(
   }
 
   def isCachedColumnBuffersLoaded: Boolean = {
-    _cachedColumnBuffers != null
+    _cachedColumnBuffers != null && isCachedRDDLoaded
+  }
+
+  def isCachedRDDLoaded: Boolean = {
```

Comment:
I'm not sure if it's necessary or not, but it's not difficult to apply double lock here and make this method thread-safe.

Reply:
Seems unnecessary. This method only writes `_cachedColumnBuffersAreLoaded` to true, so there's not a problem where it gets overwritten to false.

Reply:
OK, shall we at least mark `_cachedColumnBuffersAreLoaded` as `volatile`? In case two threads calling `isCachedRDDLoaded` at the same time.

Reply:
Makes sense. Added